### ISSUE TITLE
Unable to add column using name of previously dropped generated column

### ISSUE THREAD

**Issue by hlcianfagna** [crate/crate#16972]
### CrateDB version

5.9.2 and crate/crate:nightly-5.10.0-2024-11-12-00-02

### CrateDB setup information

Single node

### Problem description

It seems a column name that was previously used for a generated column cannot be reused, if the dropped column was not a generated column the re-adding of the column succeeds.

### Steps to Reproduce

```sql
CREATE TABLE test(a int,b int GENERATED ALWAYS AS (a+1));
REFRESH TABLE test;
ALTER TABLE test DROP COLUMN b;
ALTER TABLE test ADD COLUMN b int;
```

### Actual Result

```
SQLParseException[The table doc.test already has a column named b]

io.crate.exceptions.SQLParseException: The table doc.test already has a column named b
        at io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
        at io.crate.exceptions.SQLExceptions.prepareForClientTransmission(SQLExceptions.java:200)
        at io.crate.rest.action.SqlHttpHandler.sendResponse(SqlHttpHandler.java:176)
        at io.crate.rest.action.SqlHttpHandler.lambda$channelRead0$0(SqlHttpHandler.java:130)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:907)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:931)
        at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2401)
        at io.crate.rest.action.SqlHttpHandler.channelRead0(SqlHttpHandler.java:128)
        at io.crate.rest.action.SqlHttpHandler.channelRead0(SqlHttpHandler.java:87)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.crate.protocols.http.HttpBlobHandler.channelRead0(HttpBlobHandler.java:159)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.crate.auth.HttpAuthUpstreamHandler.handleHttpChunk(HttpAuthUpstreamHandler.java:158)
        at io.crate.auth.HttpAuthUpstreamHandler.channelRead0(HttpAuthUpstreamHandler.java:95)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:289)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
        at org.elasticsearch.transport.netty4.Netty4InboundStatsHandler.channelRead(Netty4InboundStatsHandler.java:99)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1357)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:868)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: java.lang.IllegalArgumentException: The table doc.test already has a column named b
        at io.crate.analyze.TableElementsAnalyzer$PeekColumns.visitAddColumnDefinition(TableElementsAnalyzer.java:527)
        at io.crate.analyze.TableElementsAnalyzer$PeekColumns.visitAddColumnDefinition(TableElementsAnalyzer.java:491)
        at io.crate.sql.tree.AddColumnDefinition.accept(AddColumnDefinition.java:89)
        at io.crate.analyze.TableElementsAnalyzer.analyze(TableElementsAnalyzer.java:445)
        at io.crate.analyze.AlterTableAddColumnAnalyzer.analyze(AlterTableAddColumnAnalyzer.java:55)
        at io.crate.analyze.Analyzer$AnalyzerDispatcher.visitAlterTableAddColumnStatement(Analyzer.java:295)
        at io.crate.analyze.Analyzer$AnalyzerDispatcher.visitAlterTableAddColumnStatement(Analyzer.java:259)
        at io.crate.sql.tree.AlterTableAddColumn.accept(AlterTableAddColumn.java:73)
        at io.crate.analyze.Analyzer.analyze(Analyzer.java:241)
        at io.crate.session.Session.analyze(Session.java:344)
        at io.crate.session.Session.parse(Session.java:334)
        at io.crate.rest.action.SqlHttpHandler.executeSimpleRequest(SqlHttpHandler.java:249)
        at io.crate.rest.action.SqlHttpHandler.handleSQLRequest(SqlHttpHandler.java:220)
        at io.crate.rest.action.SqlHttpHandler.channelRead0(SqlHttpHandler.java:127)
        ... 50 more
```

### Expected Result

```
OK
```

**Comment by matriv** [comment]
Seems that the table is not dropped in the first place:
```
cr> insert into test(a) values(1);
INSERT OK, 1 row affected (0.089 sec)
cr> refresh table test;
REFRESH OK, 1 row affected (0.029 sec)
cr> select * from test;
+---+---+
| a | b |
+---+---+
| 1 | 2 |
+---+---+
SELECT 1 row in set (0.026 sec)
```

**Comment by BaurzhanSakhariev** [comment]
We cannot drop columns, used in generated columns. 

https://cratedb.com/docs/crate/reference/en/latest/sql/statements/alter-table.html#drop-column

That's rather looks like drop column falsely reported success even though it was not dropped. 

So what we need here is to make ALTER TABLE to show an error

**Comment by hlcianfagna** [comment]
The documentation says we cannot drop columns which are referenced by a generated column, not the generated column itself, maybe that needs amending?

**Comment by BaurzhanSakhariev** [comment]
>The documentation says we cannot drop columns which are referenced by a generated column

Right.

>maybe that needs amending?

That depends on what we want to  do: don't allow it and fix ALTER TABLE to show an error + document it or actually allow.

We shouldn't allow dropping generated columns if it's used in partitioned by, but we already document that we cannot drop partition by columns.

**Comment by matriv** [comment]
There is a bug, the GeneratedColumn returns `false` for `isDropped()`

**Comment by matriv** [comment]
@hlcianfagna thx for reporting the issue, it has been fixed and will be available with 5.9.3 hotfix release.
